messenger.go

```diff
@@ -0,0 +1,21 @@
+package plugin
+
+// Message describes a message to be send by MessageHandler#SendMessage
+type Message struct {
+	Message  string
+	Title    string
+	Priority int
+}
+
+// MessageHandler consists of message callbacks to be used by plugins.
+type MessageHandler interface {
+	// SendMessage sends a message with the given information in the request.
+	SendMessage(msg Message) error
+}
+
+// Messenger is the interface plugins should implement to send messages.
+type Messenger interface {
+	Plugin
+	// SetMessageHandler is called every time the plugin is initialized. Plugins should record the handler and use the callbacks provided in the handler to send messages.
```

Comment:
Comments like this could be broken into multiple lines, otherwise they are pretty long. (there are multiple comments like this.